JsonSettings.Library/JsonProtectAttribute.cs

```diff
@@ -30,8 +30,23 @@ protected override IList<JsonProperty> CreateProperties(Type type, MemberSeriali
             foreach (JsonProperty prop in props.Where(p => p.PropertyType.Equals(typeof(string))))
             {
                 PropertyInfo pi = type.GetProperty(prop.UnderlyingName);
+                // Skip if member is not a property.
+                if (pi is null)
+                {
+                    continue;
+                }
+
+                // Skip if JsonIgnore.
+                if (pi.GetCustomAttribute<JsonIgnoreAttribute>() != null)
```

Comment:
You could just use the `Ignored` property that has already done this and cached it so you don't repeat the work.
```cs
if (pi.Ignored)
{
    continue;
}
```

Reply:
Yes, that would be a better implementation. If you want to make a pull request.